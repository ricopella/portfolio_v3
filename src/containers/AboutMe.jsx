import React from 'react'
import styled from '@emotion/styled'
import Page from '../components/page'
import { colors, fonts, shadows } from '../styles/variables'
import TerminalCommand from '../components/TerminalCommand'

const TerminalContainer = styled.div`
  display: grid;
  grid-template-rows: 30px max-content;
  grid-template-columns: 1fr;
  max-width: 900px;
  align-self: center;
  justify-self: center;
  box-shadow: ${shadows.box};
`
const TerminalHeaderTitle = styled.div`
  color: ${colors.gray.dark};
  align-self: center;
  justify-self: center;
`

const TerminalHeader = styled.div`
  background-color: ${colors.terminalHeader};
  border-radius: 10px 10px 0 0;
  max-height: 30px;
  width: 100%;
  display: grid;
  grid-template-columns: max-content 1fr;
`

const TerminalBody = styled.div`
  background-color: ${colors.terminalBackground};
  border-radius: 0 0 4px 4px;
  box-shadow: 0 50px 100px rgba(50, 50, 93, 0.15),
    0 15px 35px rgba(50, 50, 93, 0.2), 0 5px 15px rgba(0, 0, 0, 0.12);
  font-family: ${fonts.monospace};
  width: 100%;
  height: 100%;
  padding: 2rem 2rem;
`

const TerminalHeaderActionContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(3, max-content);
  grid-template-rows: 1fr;
  grid-column-gap: 5px;
  height: 30px;
  align-items: center;
  padding-left: 10px;
`

const TerminalHeaderActionBalls = styled.div`
  height: 15px;
  width: 15px;
  border-radius: 50%;
`

const AboutMeMain = () => (
  <Page
    id="about-me"
    style={{
      height: '100vh',
      width: '100vw',
      backgroundColor: colors.header,
      display: 'grid',
      gridTemplateColumns: '1fr',
      gridTemplateRows: 'max-content max-content',
      gridRowGap: '60px',
      alignContent: 'center',
      justifyContent: 'center',
    }}
  >
    <h1
      style={{
        fontSize: '3rem',
        textAlign: 'center',
        textDecoration: 'underline',
        color: colors.gray.dark,
      }}
    >
      {`about me`.toUpperCase()}
    </h1>
    <TerminalContainer>
      <TerminalHeader>
        <TerminalHeaderActionContainer>
          <TerminalHeaderActionBalls
            style={{ backgroundColor: colors.terminalActionClose }}
          />
          <TerminalHeaderActionBalls
            style={{ backgroundColor: colors.terminalActionMinimize }}
          />
          <TerminalHeaderActionBalls
            style={{ backgroundColor: colors.terminalActionExpand }}
          />
        </TerminalHeaderActionContainer>
        <TerminalHeaderTitle>
          narin@Narin-Portfolio: ~/About_Me
        </TerminalHeaderTitle>
      </TerminalHeader>
      <TerminalBody>
        <TerminalCommand
          question={'current_location'}
          answer={'"Los Angeles, CA"'}
        />
        <TerminalCommand
          question={'contact_information'}
          answer={'nrs710@gmail.com'}
          isLink={'mailto:nrs710@gmail.com'}
        />
        <TerminalCommand
          question={'roles'}
          answer={
            '["Front-End Web Developer", "Freelancer", "Certified Scrum Master"]'
          }
        />
        <TerminalCommand
          question={'types_of_work'}
          answer={
            '["Complex web-apps", "Data visualizations in the browser", "Small business websites", "Shopify customizations", "Mobile apps"]'
          }
        />
        <TerminalCommand
          question={'most_recent_tech_stack'}
          answer={
            '["TypeScript", "React", "Highcharts", "PostCSS", "Jest/Enzyme"]'
          }
        />
        <TerminalCommand
          question={'interests || hobbies'}
          answer={
            '["Turntablism", "Record Collecting", "Music Production", Olde English Bulldog owner", "Photography", "Boston Sports - Bruins, Red Sox, Patriots, Celtics", "Eating new places", "Coffee", "Tacos"]'
          }
        />
        <TerminalCommand />
      </TerminalBody>
    </TerminalContainer>
  </Page>
)

export default AboutMeMain
