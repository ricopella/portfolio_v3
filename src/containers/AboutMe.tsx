import React from 'react'
import styled from '@emotion/styled'
import Page from '../components/Page'
import { borders, shadows, colors, fonts } from '../styles/variables'
import TerminalCommand from '../components/TerminalCommand'

const TerminalContainer = styled.div`
  display: grid;
  grid-template-rows: 30px max-content;
  grid-template-columns: 1fr;
  max-width: 900px;
  align-self: center;
  justify-self: center;
  box-shadow: ${shadows.box};
`
const TerminalHeaderTitle = styled.div`
  color: ${colors.gray.dark};
  align-self: center;
  justify-self: center;
`

const TerminalHeader = styled.div`
  background-color: ${colors.terminalHeader};
  border-radius: ${borders.large};
  max-height: 30px;
  width: 100%;
  display: grid;
  grid-template-columns: max-content 1fr;
`

const TerminalBody = styled.div`
  background-color: ${colors.terminalBackground};
  border-radius: ${borders.medium};
  box-shadow: ${shadows.box};
  font-family: ${fonts.monospace};
  width: 100%;
  height: 100%;
  padding: 2rem 2rem;
`

const TerminalHeaderActionContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(3, max-content);
  grid-template-rows: 1fr;
  grid-column-gap: 5px;
  height: 30px;
  align-items: center;
  padding-left: 10px;
`

const TerminalHeaderActionBalls = styled.div`
  height: 15px;
  width: 15px;
  border-radius: 50%;
`

const AboutMeMain = () => (
  <Page
    id="about-me"
    style={{
      height: '100vh',
      width: '100vw',
      backgroundColor: colors.header,
      display: 'grid',
      gridTemplateColumns: '1fr',
      gridTemplateRows: 'max-content max-content',
      gridRowGap: '60px',
      alignContent: 'center',
      justifyContent: 'center',
    }}
  >
    <h1
      style={{
        fontSize: '3rem',
        textAlign: 'center',
        textDecoration: 'underline',
        color: colors.gray.dark,
      }}
    >
      {`about me`.toUpperCase()}
    </h1>
    <TerminalContainer>
      <TerminalHeader>
        <TerminalHeaderActionContainer>
          <TerminalHeaderActionBalls
            style={{ backgroundColor: colors.terminalActionClose }}
          />
          <TerminalHeaderActionBalls
            style={{ backgroundColor: colors.terminalActionMinimize }}
          />
          <TerminalHeaderActionBalls
            style={{ backgroundColor: colors.terminalActionExpand }}
          />
        </TerminalHeaderActionContainer>
        <TerminalHeaderTitle>
          admin@nrs-portfolio: ~/About_Me
        </TerminalHeaderTitle>
      </TerminalHeader>
      <TerminalBody>
        <TerminalCommand
          question={'current_location'}
          answer={'"Los Angeles, CA"'}
        />
        <TerminalCommand
          question={'contact_information'}
          answer={'nrs710@gmail.com'}
          isLink={'mailto:nrs710@gmail.com'}
        />
        <TerminalCommand
          question={'roles'}
          answer={
            '["Front-End Web Developer", "Freelancer", "Certified Scrum Master"]'
          }
        />
        <TerminalCommand
          question={'types_of_work'}
          answer={
            '["Complex web-apps", "Data visualizations in the browser", "Small business websites", "Shopify customizations", "Mobile apps"]'
          }
        />
        <TerminalCommand
          question={'most_recent_tech_stack'}
          answer={
            '["TypeScript", "React", "Highcharts", "PostCSS", "Jest/Enzyme"]'
          }
        />
        <TerminalCommand
          question={'interests || hobbies'}
          answer={
            '["Turntablism", "Record Collecting", "Music Production", Olde English Bulldog owner", "Photography", "Boston Sports - Bruins, Red Sox, Patriots, Celtics", "Eating new places", "Coffee", "Tacos"]'
          }
        />
        <TerminalCommand />
      </TerminalBody>
    </TerminalContainer>
  </Page>
)

export default AboutMeMain
