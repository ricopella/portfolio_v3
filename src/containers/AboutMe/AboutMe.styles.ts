import styled from '@emotion/styled'
import { css } from '@emotion/core'
import { motion } from 'framer-motion'
import { OutboundLink } from 'gatsby-plugin-google-analytics'
import {
  BREAKPOINTS,
  BORDERS,
  SHADOWS,
  COLORS,
  FONTS,
} from '../../styles/variables'

const TerminalContainer = styled(motion.div)`
  display: grid;
  grid-template-rows: 1.875rem max-content;
  grid-template-columns: 1fr;
  max-width: 56.25rem;
  align-self: center;
  justify-self: center;
  box-shadow: ${SHADOWS.box};
  margin-top: 1rem;
`
const TerminalHeaderTitle = styled.div`
  color: ${COLORS.gray.dark};
  align-self: center;
  justify-self: center;
`

const TerminalHeader = styled.div`
  background-color: ${COLORS.terminalHeader};
  border-radius: ${BORDERS.large};
  max-height: 1.875rem;
  width: 100%;
  display: grid;
  grid-template-columns: max-content 1fr;
`

const TerminalBody = styled.div`
  background-color: ${COLORS.terminalBackground};
  border-radius: ${BORDERS.medium};
  box-shadow: ${SHADOWS.box};
  font-family: ${FONTS.monospace};
  width: 100%;
  height: 100%;
  padding: 2rem 2rem;
`

const TerminalHeaderActionContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(3, max-content);
  grid-template-rows: 1fr;
  grid-column-gap: 0.3125rem;
  height: 1.875rem;
  align-items: center;
  padding-left: 0.625rem;
`

const TerminalHeaderActionBalls = styled.div`
  height: 0.9375rem;
  width: 0.9375rem;
  border-radius: 50%;
`
const BioContainer = styled(motion.div)`
  display: grid;
  grid-template-columns: max-content 1fr;
  grid-template-rows: repeat(3, max-content);
  grid-column-gap: 1.25rem;
  margin-top: 1rem;

  @media (max-width: 1244px) {
    grid-template-columns: 1fr;
  }

  @media (max-width: ${BREAKPOINTS.mdRem}) {
    justify-items: center;
    max-width: calc(56.25rem - 10%);
    margin: 0 0;
  }
`

const BioHeadingWrapper = styled.div`
  display: grid;
  grid-template-columns: repeat(2, max-content);
  grid-template-rows: max-content;
  grid-column-gap: 1.25rem;
  align-items: center;
  margin-bottom: 1rem;

  @media (max-width: ${BREAKPOINTS.mdRem}) {
    grid-template-columns: 1fr;
    justify-items: center;
  }
`

const BioHeadingText = styled.div`
  max-width: 15.625rem;
  color: ${COLORS.gray.light};

  & h2,
  & h4 {
    color: ${COLORS.gray.light};
  }
`
const BioParagraph = styled('p')`
  color: ${COLORS.gray.light};
  grid-column: 1 / -1;

  &.first {
    grid-column: 2;
    align-self: end;

    @media (max-width: 1244px) {
      grid-column: 1 / -1;
    }
  }
`

const PageHeading = css`
  fontsize: 3rem;
  text-align: center;
  text-decoration: underline;
  color: ${COLORS.gray.calm};
`

const ContactLink = styled(OutboundLink)`
  color: ${COLORS.terminalBackground};

  :hover {
    color: ${COLORS.gray.calm};
    transform: scale(1.1);
  }
`

export default {
  ContactLink,
  BioContainer,
  BioHeadingText,
  BioHeadingWrapper,
  BioParagraph,
  PageHeading,
  TerminalBody,
  TerminalContainer,
  TerminalHeader,
  TerminalHeaderActionBalls,
  TerminalHeaderActionContainer,
  TerminalHeaderTitle,
}
