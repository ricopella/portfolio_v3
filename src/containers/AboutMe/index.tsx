import Img from 'gatsby-image'
import Page from '../../components/Page/index'
import ProfileImage from '../../components/ProfileImage'
import React, { useState } from 'react'
import Styled from './AboutMe.styles'
import TerminalCommand from '../../components/TerminalCommand/index'
import ToggleSwitch from '../../components/ToggleSwitch'
import useSiteMetaData from '../../hooks/useSiteMetaData'
import useSpotifyData from '../../hooks/useSpotifyData'
import { AboutMeViewsArr } from '../../types'
import { COLORS } from '../../styles/variables'
import { SLIDE_IN_ANIMATION_OPTIONS } from '../../styles/variables'

const VIEWS: AboutMeViewsArr = ['bio', 'terminal']
const HEADING = 'About Me'

const AboutMeMain = () => {
  const { terminalCommands } = useSiteMetaData()
  const [currentView, setCurrentView] = useState<string>(VIEWS[0])
  const recentTracks = useSpotifyData()
  console.log(recentTracks)

  const renderTerminalHeader = () => (
    <Styled.TerminalHeader>
      <Styled.TerminalHeaderActionContainer>
        <Styled.TerminalHeaderActionBalls
          style={{ backgroundColor: COLORS.terminalActionClose }}
        />
        <Styled.TerminalHeaderActionBalls
          style={{ backgroundColor: COLORS.terminalActionMinimize }}
        />
        <Styled.TerminalHeaderActionBalls
          style={{ backgroundColor: COLORS.terminalActionExpand }}
        />
      </Styled.TerminalHeaderActionContainer>
      <Styled.TerminalHeaderTitle>
        admin@nrs-portfolio: ~/About_Me
      </Styled.TerminalHeaderTitle>
    </Styled.TerminalHeader>
  )

  const renderTerminal = () => (
    <Styled.TerminalContainer {...SLIDE_IN_ANIMATION_OPTIONS}>
      {renderTerminalHeader()}
      <Styled.TerminalBody>
        {terminalCommands.map(command => (
          <TerminalCommand
            key={`terminal_command_${command.question}`}
            answer={command.answer}
            href={command.href}
            question={command.question}
          />
        ))}
        <TerminalCommand />
      </Styled.TerminalBody>
    </Styled.TerminalContainer>
  )

  const renderInfo = () => (
    <Styled.BioContainer {...SLIDE_IN_ANIMATION_OPTIONS}>
      <Styled.BioHeadingWrapper>
        <ProfileImage />
        <Styled.BioHeadingText>
          <h2>{`Hi, I'm Narin.`}</h2>
          <h4>{`I'm grateful that you've visited my page.`}</h4>
        </Styled.BioHeadingText>
      </Styled.BioHeadingWrapper>
      <Styled.BioParagraph className="first">
        {`Iâ€™m a frontend software engineer & freelance web developer living in Los
        Angeles. I'm passionate about music, sports, technology and healthy
        living. Collectively I have 10+ years experience in the Sports,
        Entertainment, and Events industries. I'm a life-long learner that's
        looking for ways to get out of my comfort zone to grow my versatility as
        a developer.`}
      </Styled.BioParagraph>
      <Styled.BioParagraph>
        {`I've built sites ranging from simple portfolios, eCommerce shops, to
        data driven B2B web apps with machine learning and data visualizations.
        Some of my freelance web development experience has been with Shopify,
        Wix, and SquareSpace. I've also worked on a freelance project that
        designed a custom API that integrated with 3rd party APIs to automate
        inventory and sales.`}
      </Styled.BioParagraph>
      <Styled.BioParagraph>
        {`In 2017, I attended UCLA's Bootcamp to accelerate my learning and gain
        knowledge of the Node.js ecosystem. Since then, I fell in love with &
        have been primarily using React in web applications. For static sites, I
        love using Gatsby for creating blazing fast, SEO-driven headless
        frontend user interfaces (like this site).`}
      </Styled.BioParagraph>
      <Styled.BioParagraph>
        {`When I'm not looking into a screen filled with code, I'm usually walking
        with my dog, at the gym, meditating, trying a new plant based recipe, or
        doing some form of musical expression. I've owned a pair of turntables,
        beat machine, and records since I was in the 9th grade. I'm in a DJ
        group called `}
        <Styled.ContactLink
          href={'https://themusicplease.com/'}
          target="_blank"
          rel="noopener noreferrer"
        >
          {`Music Please`}
        </Styled.ContactLink>
        {` which started from a radio show that lasted 7 years. And, I'm an avid
        New England sports fan!`}
      </Styled.BioParagraph>
      <div>
        {recentTracks.map((audio, idx) => (
          <div key={audio.track.name + idx}>
            {audio.track.name} -{' '}
            <a href={audio.track.href}>{audio.track.href} </a>
            {audio.track.artists.map(x => `${x.name} -`)}
            {/* <Img fluid={audio.track.image.localFile.childImageSharp.fluid} /> */}
          </div>
        ))}
      </div>
    </Styled.BioContainer>
  )

  return (
    <Page id="about-me" title={HEADING}>
      <ToggleSwitch
        currentView={currentView}
        setView={setCurrentView}
        views={VIEWS}
      />
      {currentView === 'bio' ? renderInfo() : renderTerminal()}
    </Page>
  )
}

export default AboutMeMain
