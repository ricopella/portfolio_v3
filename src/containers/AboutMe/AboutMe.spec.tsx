import renderer from 'react-test-renderer'
import React from 'react'
import AboutMe from './index'
import { render } from '@testing-library/react'

describe('<AboutMe />', () => {
  it('should match AboutMe snapshot', () => {
    const internalTree = renderer.create(<AboutMe />).toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should render children', () => {
    const { getByText } = render(<AboutMe />)

    expect(getByText(/About Me/i)).toBeTruthy()
  })
})
