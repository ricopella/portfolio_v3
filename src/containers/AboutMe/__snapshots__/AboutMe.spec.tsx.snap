// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutMe /> should match AboutMe snapshot 1`] = `
<div
  className="css-1ivi42b"
  id="about-me"
>
  <h1
    css={
      Object {
        "map": undefined,
        "name": "2guzxi",
        "next": undefined,
        "styles": "
  fontsize: 3rem;
  text-align: center;
  text-decoration: underline;
  color: hsla(270, 17.119554496%, 0%, 0.92);
",
        "toString": [Function],
      }
    }
  >
    ABOUT ME
  </h1>
  <div
    className="css-ka0t2p"
  >
    <div
      className="css-h0tjk2"
    >
      <div
        className="css-1snu6sh"
      >
        <div
          className="css-trjo67"
          style={
            Object {
              "backgroundColor": "#f96256",
            }
          }
        />
        <div
          className="css-trjo67"
          style={
            Object {
              "backgroundColor": "#fdbc3d",
            }
          }
        />
        <div
          className="css-trjo67"
          style={
            Object {
              "backgroundColor": "#2dbb41",
            }
          }
        />
      </div>
      <div
        className="css-1ih0lmc"
      >
        admin@nrs-portfolio: ~/About_Me
      </div>
    </div>
    <div
      className="css-hw2mt4"
    >
      <div
        className="css-3yu43t"
      >
        <div
          className="css-ntk5lm"
        >
          <span
            className="css-zbjqpz"
          >
            &gt;&gt; 
          </span>
           current_location
        </div>
        <div
          className="css-1yzmtjz"
        >
          "Los Angeles, CA"
        </div>
      </div>
      <div
        className="css-3yu43t"
      >
        <div
          className="css-ntk5lm"
        >
          <span
            className="css-zbjqpz"
          >
            &gt;&gt; 
          </span>
           contact_information
        </div>
        <div
          className="css-1yzmtjz"
        >
          <a
            className="css-1t2r17y"
            href="mailto:nrs710@gmail.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            "nrs710@gmail.com"
          </a>
        </div>
      </div>
      <div
        className="css-3yu43t"
      >
        <div
          className="css-ntk5lm"
        >
          <span
            className="css-zbjqpz"
          >
            &gt;&gt; 
          </span>
           roles
        </div>
        <div
          className="css-1yzmtjz"
        >
          ["Front-End Web Developer", "Freelancer", "Certified Scrum Master"]
        </div>
      </div>
      <div
        className="css-3yu43t"
      >
        <div
          className="css-ntk5lm"
        >
          <span
            className="css-zbjqpz"
          >
            &gt;&gt; 
          </span>
           types_of_work
        </div>
        <div
          className="css-1yzmtjz"
        >
          ["Complex web-apps", "Data visualizations in the browser", "Small business websites", "Shopify customizations", "Mobile apps"]
        </div>
      </div>
      <div
        className="css-3yu43t"
      >
        <div
          className="css-ntk5lm"
        >
          <span
            className="css-zbjqpz"
          >
            &gt;&gt; 
          </span>
           most_recent_tech_stack
        </div>
        <div
          className="css-1yzmtjz"
        >
          ["TypeScript", "React", "Highcharts", "PostCSS", "Jest/Enzyme"]
        </div>
      </div>
      <div
        className="css-3yu43t"
      >
        <div
          className="css-ntk5lm"
        >
          <span
            className="css-zbjqpz"
          >
            &gt;&gt; 
          </span>
           interests || hobbies
        </div>
        <div
          className="css-1yzmtjz"
        >
          ["Turntablism", "Record Collecting", "Music Production", Olde English Bulldog owner", "Photography", "Boston Sports - Bruins, Red Sox, Patriots, Celtics", "Eating new places", "Coffee", "Tacos"]
        </div>
      </div>
      <div
        className="css-3yu43t"
      >
        <div
          className="css-ntk5lm"
        >
          <span
            className="css-zbjqpz"
          >
            &gt;&gt; 
          </span>
          <div
            className="css-1odt7h0"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
